src/ronin/profile/ProfileXComponents.sol

```diff
@@ -9,6 +9,29 @@ import "./ProfileHandler.sol";
 pragma solidity ^0.8.9;
 
 abstract contract ProfileXComponents is IProfile, ProfileHandler {
+  /**
+   * @inheritdoc IProfile
+   */
+  function execCreateRollup(address id, uint32 rollupId) external onlyContract(ContractType.ZK_ROLLUP_MANAGER) {
+    CandidateProfile storage _profile = _getId2ProfileHelper(id);
+
+    _requireNotOnRenunciation(id);
+
+    if (_registry[rollupId]) revert ErrRollupIdAlreadyRegistered(rollupId);
+    if (rollupId == 0) revert ErrZeroRollupId(id);
+    if (_profile.id == address(0)) revert ErrNonExistentProfile();
```

Comment:
nit: this is not needed, as there is a check in _getId2ProfileHelper already

Reply:
fixed.